proto/spire/api/server/federationmanagement/v1/federationmanagement.proto

```diff
@@ -0,0 +1,154 @@
+syntax = "proto3";
+package spire.api.server.federationmanagement.v1;
+option go_package = "github.com/spiffe/spire-api-sdk/proto/spire/api/server/federationmanagement/v1;federationmanagementv1";
+
+import "google/protobuf/empty.proto";
+import "spire/api/types/federationrelationship.proto";
+import "spire/api/types/status.proto";
+
+// Manages the federation relationships with foreign trust domains.
+service FederationManagement {
+    // Lists federation relationships with foreign trust domains.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc ListFederationRelationships(ListFederationRelationshipsRequest) returns (ListFederationRelationshipsResponse);
+
+    // Gets a federation relationship with a foreign trust domain.
+    // If there is no federation relationship with the specified
+    // trust domain, NOT_FOUND is returned.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc GetFederationRelationship(GetFederationRelationshipRequest) returns (spire.api.types.FederationRelationship);
+
+    // Batch creates one or more federation relationships with
+    // foreign trust domains.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc BatchCreateFederationRelationship(BatchCreateFederationRelationshipRequest) returns (BatchCreateFederationRelationshipResponse);
+
+    // Batch updates one or more federation relationships with
+    // foreign trust domains.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc BatchUpdateFederationRelationship(BatchUpdateFederationRelationshipRequest) returns (BatchUpdateFederationRelationshipResponse);
+
+    // Batch deletes federation relationships with foreign trust domains.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc BatchDeleteFederationRelationship(BatchDeleteFederationRelationshipRequest) returns (BatchDeleteFederationRelationshipResponse);
+
+    // Refreshes the bundle from the specified federated trust domain.
+    // If there is not a federation relationship configured with the
+    // specified trust domain, NOT_FOUND is returned.
+    //
+    // The caller must be local or present an admin X509-SVID.
+    rpc RefreshBundle(RefreshBundleRequest) returns (google.protobuf.Empty);
```

Comment:
This one feels a little inconsistent with the rest. Maybe `ReloadFederationRelationship` could work?

Reply:
How do we envision this working? It will trigger a refresh/reload on the server handling this RPC? It will be blocking or non-blocking?

Reply:
>This one feels a little inconsistent with the rest. Maybe ReloadFederationRelationship could work?

The reason behind the name is that the primary goal is really to refresh the bundle from a specified trust domain. We currently log a message at Info level saying `Bundle refreshed` when a bundle has been successfully refreshed and seems natural to call it `RefreshBundle`.
I see your point about consistency with other RPCs, but my fear is that `ReloadFederationRelationship` may induce to think that there is an attribute of the federation relationship itself that you want to refresh, e.g. the bundle endpoint profile used or the URL and may not be obvious that the bundle will be refreshed.
Having said that, I'm not strongly opposed to change the name to `ReloadFederationRelationship` since it also has a logic.

Reply:
>How do we envision this working? It will trigger a refresh/reload on the server handling this RPC? It will be blocking or non-blocking?

The use case that I saw for this RPC is: TD1 has a federation relationship with TD2. An operator or automated process in TD1 knows that the bundle of TD2 has changed for whatever reason and the next bundle refresh is scheduled to happen in days, which is not acceptable. The RPC is called in TD1 to refresh the bundle from TD2 which would effectively do the same as what would happen in the next scheduled refresh, i.e. set the updated bundle of TD2 in TD1. I envisioned this as a blocking call as an easy way to know that the operation completed, but I'm open to suggestions.